apps/server/src/shared/domain/entity/submission.entity.ts

```diff
@@ -112,10 +115,13 @@ export class Submission extends BaseEntityWithTimestamps {
 	// Bad that the logic is needed to expose the userIds, but is used in task for now.
 	// Check later if it can be replaced and remove all related code.
 	public getSubmitterIds(): EntityId[] {
-		const creatorId = this.student.id;
+		const creatorId = this.student?.id ?? undefined;
```

Comment:
Here is a problem I think, the student is not populate.
this.student is an ObejectId.
Please check it

Reply:
No student is user per code

Reply:
look at line 36-37

Reply:
I mean it is possible that student exists but not populate
If you call repo.findAllByTaskIds() than in line 27 await this.populateReferences(submissions); is called and in
 populateReferences() hasn't user.populate
 
 so 
 const creatorId = this.student?.id
 
 is creatorId === undifined
 
 but so 
 const creatorId = this.student
 
  is creatorId === ObjectId("xxx")

Reply:
Sorry but I don't understand what do you mean exactly could you expand your thought?
If we call this.studnet creatorId will be type User 
Before making it optional this.student.id was called to get the userId

Reply:
Sorry i have checked it, it works.